### PR TITLE
Physics

### LINKED ISSUE #16: Physics?
We need physics in this game like so stuff can fall and shit

EDIT(lambdaviking): This entails adding velocity (momentum) to entities and gravity.

### PR DESCRIPTION
**Pull request by tojaroslaw**
resolves #16
we have physics now

### CODE REVIEW

File: src/snorri/entities/Entity.java [hunk 1]
```diff
@@ -197,6 +202,17 @@ public boolean intersectsWall(World world) {
 		return false;	
 	}
 	
+	public boolean willIntersectSurfaceY(World world, Vector newPos) {
```
[the remark this answers is not in the document]
Comment (tojaroslaw) [reply src/snorri/entities/Entity.java]: it's a modified version that just checks if you're hitting a floor, if you look up a few lines, intersectsWall is still there

File: src/snorri/entities/Entity.java [hunk 2]
```diff
@@ -226,11 +242,17 @@ public String toStringDebug() {
...
+	 * 
+	 * @param world
+	 * @param deltaTime
+	 */
+	public void update(World world, double deltaTime) {
 		if (!hasCycled && (animation == null || animation.hasCycled())) {
 			onCycleComplete(world);
 			hasCycled = true;
 		}
+		setPos(pos.add(velocity.multiply(deltaTime))); // XXX this has the potential to cause some bugs
```
[the remark this answers is not in the document]
Comment (tojaroslaw) [reply src/snorri/entities/Entity.java]: this is only an issue if we want gravity to apply to something that soesn't interact with surfaces

File: src/snorri/world/TileType.java [hunk 3]
```diff
@@ -42,8 +42,8 @@ public T getValue() {
 			return value;
 		}
 		
-		public static Param<Boolean> pathable(Boolean value) {
-			return new Param<>(Key.PATHABLE, value);
+		public static Param<Boolean> isSurface(Boolean value) {
```
[the remark this answers is not in the document]
Comment (tojaroslaw) [reply src/snorri/world/TileType.java]: isPathable has meant the opposite of what we want it to mean now, so i wanted to make it less ambiguous

File: src/snorri/entities/Entity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (viking-sudo-rm) [reply src/snorri/entities/Entity.java]: Okay. Can you make the name more descriptive?

File: src/snorri/entities/Entity.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (viking-sudo-rm) [reply src/snorri/entities/Entity.java]: What do you mean by "interact with surfaces"? This is an issue exactly when something can fall into a surface.

File: src/snorri/world/TileType.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (viking-sudo-rm) [reply src/snorri/world/TileType.java]: Renaming it makes sense.

My point was that I don't think `isSurface` should be a property of the `TileType`. Whether or not a tile is a surface depends really on the surrounding tile, not the type of tile. Therefore, we should get rid of the property on tile objects and instead call something like the method above, which could go in `TileLayer`.

**Pull request merged**